File: src/Worker/content/NBB.Worker/Messaging/SampleSubscriberPipelineMiddleware.cs
using Microsoft.Extensions.Logging;
using NBB.Core.Pipeline;
using NBB.Messaging.Abstractions;
using System;
using System.Threading;
using System.Threading.Tasks;

namespace NBB.Worker.Messaging
{
    public class SampleSubscriberPipelineMiddleware: IPipelineMiddleware<MessagingContext>
    {
        private readonly ILogger<SampleSubscriberPipelineMiddleware> _logger;

        public SampleSubscriberPipelineMiddleware(ILogger<SampleSubscriberPipelineMiddleware> logger)
        {
            _logger = logger;
        }

        public async Task Invoke(MessagingContext context, CancellationToken cancellationToken, Func<Task> next)
        {
            _logger.LogDebug("Message {@Message} was received.", context.MessagingEnvelope.Payload);
            await next();
        }
    }
}

File: src/Worker/NBB.Worker.FSharp/content/NBB.Worker/Messaging/SampleSubscriberPipelineMiddleware.fs
namespace NBB.Worker.Messaging

open Microsoft.Extensions.Logging
open NBB.Core.Pipeline
open NBB.Messaging.Abstractions
open System
open System.Threading
open System.Threading.Tasks

type SampleSubscriberPipelineMiddleware(logger: ILogger<SampleSubscriberPipelineMiddleware>) =
    interface IPipelineMiddleware<MessagingContext> with
        member _.Invoke(context: MessagingContext, cancellationToken: CancellationToken, next: Func<Task>) =
            logger.LogDebug("Message {@Message} was received.", context.MessagingEnvelope.Payload)
            next.Invoke()
